packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -177,44 +450,57 @@ const Dropdown = forwardRef(
     }, [size, rtl, insideOverflowContainer, insideOverflowWithTransformContainer, extraStyles, multi, multiline]);
 
     const Menu = useCallback(
-      props => (
+      (props: CustomMenuProps) => (
```

Comment:
I wonder if we better use the MenuComponent's interface already, it's out of this scope so maybe leaving it as any for now would be better. Same for the rest sub components here